src/main/java/com/compilercharisma/chameleonbusinessstudio/controller/ScheduleController.java

```diff
@@ -1,10 +1,99 @@
 package com.compilercharisma.chameleonbusinessstudio.controller;
 
+import java.net.URI;
+import java.util.List;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import com.compilercharisma.chameleonbusinessstudio.dto.Schedule;
+import com.compilercharisma.chameleonbusinessstudio.service.ScheduleService;
+
+import reactor.core.publisher.Mono;
 
+/**
+ * Handles HTTP requests related to schedules of repeating appointments
+ */
+// only talents, admins, or organizers should access this
 @RequestMapping("/api/v1/schedules")
 @RestController
 public class ScheduleController {
-    
+    private ScheduleService schedules;
+
+    /**
+     * @param schedules handles business logic needed by this controller
+     */
+    public ScheduleController(ScheduleService schedules){
+        this.schedules = schedules;
+    }
+
+    /**
+     * @return a page containing all schedules stored in the application
+     */
+    @GetMapping()
```

Comment:
Aren't these methods missing an endpoint in the `@GetMapping()` tag?

Reply:
Nope! The `@RequestMapping` annotating the class grabs requests to `/api/v1/schedules`, then the empty `@GetMapping` annotating the method means "GET requests to this controller's route go here", so it receives `GET /api/v1/schedules`. 

`@GetMapping("/foo")` on a method means "GET requests to this controller's route **followed by /foo** go here", so it would receive `GET /api/v1/schedules/foo`.